packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -364,18 +366,8 @@ class _HourMinuteControl extends StatelessWidget {
       states,
     ).copyWith(color: effectiveTextColor);
 
-    final double height;
-    switch (_TimePickerModel.entryModeOf(context)) {
-      case TimePickerEntryMode.dial:
-      case TimePickerEntryMode.dialOnly:
-        height = defaultTheme.hourMinuteSize.height;
-      case TimePickerEntryMode.input:
-      case TimePickerEntryMode.inputOnly:
-        height = defaultTheme.hourMinuteInputSize.height;
-    }
-
```

Comment:
I suggest asserting the mode at the beginning of the `build` method of each `Dial` widget for clarity. Probably extract it into a common function.